frontend/src/app/components/employee/modals/employee-absence-edit-modal/employee-absence-edit-modal.component.ts

```diff
@@ -1,34 +1,49 @@
-import { Component, OnInit, Inject } from "@angular/core";
+import { Component, OnInit, Inject, OnDestroy } from "@angular/core";
+import { FormGroup, FormControl, Validators } from "@angular/forms";
+import { Absence, AbsenceType } from "src/app/models/Absence.model";
+import { Subject } from "rxjs";
 import { ApiCommunicationService } from "src/app/services/api-communication.service";
 import { MatDialogRef, MAT_DIALOG_DATA } from "@angular/material";
-import { FormGroup, FormControl } from "@angular/forms";
-import { AbsenceType, Absence } from "src/app/models/Absence.model";
-import { ActivatedRoute, Params } from "@angular/router";
-import { Subject } from "rxjs";
 import { takeUntil } from "rxjs/operators";
+import { User } from "src/app/models/User.model";
+import * as moment from "moment";
 
 @Component({
   selector: "app-employee-absence-edit-modal",
   templateUrl: "./employee-absence-edit-modal.component.html",
   styleUrls: ["./employee-absence-edit-modal.component.css"]
 })
-export class EmployeeAbsenceEditModalComponent implements OnInit {
-  private createAbsenceForm: FormGroup = new FormGroup({});
+export class EmployeeAbsenceEditModalComponent implements OnInit, OnDestroy {
+  private createAbsenceForm: FormGroup;
   private types;
   private error: string;
   private absence: Absence;
-  private update: boolean = false;
-  private message = "Modify";
+  private message = "Edit";
   private _unsubscribe$ = new Subject<void>();
+  private leaders: User[];
+  private users: User[];
+  private isTypeEdit = { value: false };
```

Comment:
kérdezni akartam már az előző PR-nél is de van valami oka hogy ez nem egy sima boolean hanem objektumba van ágyazva?
```suggestion
  private isTypeEdit = false;
```

Reply:
van egy onEdit(element) függvényem, ami true-ra állítja azt, amit kap a html-ről. azért objektum, hogy referencia szerint adódjon át

Reply:
Még várj egy picit, kicserélem itt is az ikonokat